core/src/test/java/com/linecorp/armeria/client/DefaultClientRequestContextTest.java

```diff
@@ -38,6 +40,10 @@ public void deriveContext() {
                 Endpoint.of("example.com", 8080), HttpMethod.POST, "/foo", null, null,
                 ClientOptions.DEFAULT, mock(Request.class));
 
+        final DefaultHttpHeaders headers = new DefaultHttpHeaders();
+        headers.set(AsciiString.of("my-header"), "baz");
+        originalCtx.setAdditionalRequestHeaders(headers);
```

Comment:
Perhaps we could try `addAdditionalRequestHeader()` and more setters and adders as well to increase test coverage?